app/controllers/data_vis_controller.rb

```diff
@@ -1,5 +1,37 @@
 class DataVisController < ApplicationController
+    def data_view
+        #overall tour information
+        @tour_name = Tour.find_by(id: 7)&.name
+        @expenses_tour = Expense.where(tour_id: 7)
 
-    def data_view; end
-    
+        #budget calculations
+        tour_budget = Tour.find_by(id: 1)&.budget 
```

Comment:
For this `id`, you will need to update the route to something like

```ruby
get "/tour/:id/analytics"
```
And then access the id in your controller using `params[:id]`. Since you use this a lot below, I suggest assigning